spec/till_spec.rb

```diff
@@ -0,0 +1,142 @@
+require 'till'
+require 'timecop'
+
+describe Till do
+
+  let(:till) { described_class.new( { shopName: "The Coffee Connection",
+                                      address: "123 Lakeside Way",
+                                      phone: "16503600708",
+                                      prices: { "Cafe Latte"=> 4.75, "Flat White"=> 4.75 }
+                                     } ) }
+
+  let(:order) { double(:order, customers: ["Jane", "John"], 
+                               items: ["Cafe Latte", "Cafe Latte", "Flat White"],
+                               table: 1 ) }
+
+  let(:amount_discount) { double(:discount, type: :amount, trigger: 10.00 , percent: 10.00) }
+  let(:line_discount) { double(:discount, type: :line, line: "Latte", percent: 5.00) }
+
+  describe '#total' do
+
+    it 'calculates the amount due' do
+      expect(till.total(order)).to eq( { amount_due: 15.48 } )
+    end
+
+    context 'when amount discount is added' do
+
+      it 'calculates amount due - discount' do
+        till.add_discount(amount_discount)
+        expect(till.total(order)).to eq( { amount_due: 13.93 } )
+      end
+
+      it 'shows discount on receipt' do
+        till.add_discount(amount_discount)
+        till.total(order)
+        expect(till.take_payment(20.00)[:discount]).to eq(1.55)
+      end
+
+    end
+
+    context 'when line discount is added' do
+
+      it 'calculates amount due - discount' do
+        till.add_discount(line_discount)
+        expect(till.total(order)).to eq( { amount_due: 15.00 } )
+      end
+
+      it 'shows discount on receipt' do
+        till.add_discount(line_discount)
+        till.total(order)
+        expect(till.take_payment(20.00)[:discount]).to eq(0.48)
+      end
+
+    end
+
+  end
+
+  describe '#take_payment' do
+
+    before do
+      till.total(order)
+    end
+
+    it 'produces receipt with time' do
+      Timecop.freeze(Time.local(2015, 10, 10, 8, 0))
```

Comment:
Rails/TimeZone: Do not use `Time.local` without zone. Use one of `Time.zone.local`, `Time.local.current`, `Time.local.in_time_zone`, `Time.local.utc`, `Time.local.getlocal`, `Time.local.iso8601`, `Time.local.jisx0301`, `Time.local.rfc3339`, `Time.local.to_i`, `Time.local.to_f` instead. (https://github.com/bbatsov/rails-style-guide#time, http://danilenko.org/2012/7/6/rails_timezones)